lib/Remote/RemoteWebElement.php

```diff
@@ -325,17 +326,33 @@ public function sendKeys($value)
     {
         $local_file = $this->fileDetector->getLocalFile($value);
         if ($local_file === null) {
-            $params = [
-                'value' => WebDriverKeys::encode($value),
-                ':id' => $this->id,
-            ];
+            if ($this->w3cCompliant) {
+                $params = [
+                    'text' => (string) $value,
+                    ':id' => $this->id,
+                ];
+            } else {
+                $params = [
+                    'value' => WebDriverKeys::encode($value),
+                    ':id' => $this->id,
+                ];
+            }
             $this->executor->execute(DriverCommand::SEND_KEYS_TO_ELEMENT, $params);
         } else {
             $remote_path = $this->upload($local_file);
-            $params = [
-                'value' => WebDriverKeys::encode($remote_path),
-                ':id' => $this->id,
-            ];
+
+            if ($this->w3cCompliant) {
```

Comment:
dtto, maybe put this in compatibility layer instead of the conditional clauses

Reply:
Same here, it will make it easier to remove support for the legacy protocol in v2.